code/__HELPERS/logging.dm

```diff
@@ -36,6 +36,22 @@
 	GLOB.STUI.admin.Add("\[[time_stamp()]]ADMIN: [text]")
 	GLOB.STUI.processing |= STUI_LOG_ADMIN
 
+/proc/check_slur(var/text, var/mob/source, var/method = "Speech")
+	if(!CONFIG_GET(flag/log_slurs))
+		return FALSE
+	var/list/TriggerWords = file2list("config/banned_words.txt")
+	for(var/Trigger in TriggerWords)
+		if(findtext(text,Trigger))
+			log_slur(source, Trigger, text)
+			return TRUE //Blocks message being sent.
```

Comment:
return the full text here as an error message but with the slur word underlined so the user understands what exactly was blocked. that way if there's a false positive it can be reported, and also silent failing isn't good UX.